### PR TITLE
feat: update otel version

### PR DESCRIPTION
**Pull request by rogerogers**
#### What type of PR is this?
feat: update otel version

#### What this PR does / why we need it (English/Chinese):
en: update otel version, update metric to new version
zh: 更新otel版本，更新metric

### CODE REVIEW

**Review by CoderPoet — APPROVED** [review]
/LGTM

File: tracing/metrics.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ var (
 		semconv.HTTPHostKey,
 		semconv.HTTPRouteKey,
 		semconv.HTTPMethodKey,
+		semconv.HTTPStatusCodeKey,
```
Comment (CoderPoet) [inline tracing/metrics.go:47]: http.status_code as an metrics label, there may be a high cardinality problem

Comment (rogerogers) [reply tracing/metrics.go:47]: 感觉还好，不会特别多，看了python和Java的自动程序也有加这个。

Comment (CoderPoet) [reply tracing/metrics.go:47]: 嗯嗯 问题确实不大，常用的 http status code 也是可枚举的，不会膨胀太厉害

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**